inference-engine/samples/speech_sample/main.cpp

```diff
@@ -923,24 +998,36 @@ int main(int argc, char *argv[]) {
                         continue;
                     }
 
-                    ptrInputBlobs.clear();
-                    for (auto& input : cInputInfo) {
-                        ptrInputBlobs.push_back(inferRequest.inferRequest.GetBlob(input.first));
-                    }
+                    if (FLAGS_iname.empty()) {
+                        for (size_t i = 0; i < numInputArkFiles; ++i) {
+                            MemoryBlob::Ptr minput = as<MemoryBlob>(ptrInputBlobs[i]);
+                            if (!minput) {
+                                slog::err << "We expect ptrInputBlobs[" << i << "] to be inherited from MemoryBlob, " <<
+                                          "but in fact we were not able to cast input blob to MemoryBlob" << slog::endl;
+                                return 1;
+                            }
+                            // locked memory holder should be alive all time while access to its buffer happens
+                            auto minputHolder = minput->wmap();
 
-                    for (size_t i = 0; i < numInputArkFiles; ++i) {
-                        MemoryBlob::Ptr minput = as<MemoryBlob>(ptrInputBlobs[i]);
-                        if (!minput) {
-                            slog::err << "We expect ptrInputBlobs[" << i << "] to be inherited from MemoryBlob, " <<
-                                "but by fact we were not able to cast input blob to MemoryBlob" << slog::endl;
-                            return 1;
+                            std::memcpy(minputHolder.as<void *>(),
+                                        inputFrame[i],
+                                        minput->byteSize());
                         }
-                        // locked memory holder should be alive all time while access to its buffer happens
-                        auto minputHolder = minput->wmap();
+                    } else {
+                        for (size_t i = 0; i < ptrInputBlobs.size(); i++) {
```

Comment:
duplicated code with 2 lines changed.

Reply:
@bsochack, can you please explain what you mean?

Reply:
From what I see, the code could be simplified to:
```suggestion
size_t num_files = FLAGS_iname.empty() ? numInputArkFiles: ptrInputBlobs.size();
for (size_t i = 0; i < num_files; ++i) {
    MemoryBlob::Ptr minput = as<MemoryBlob>(ptrInputBlobs[i]);
    if (!minput) {
    ...
```